### PR TITLE
[TS] LPS-124433 Sort widget no longer works with DDM fields

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-124433

In order to solve this issue, I have created two new `createDDMStructureFieldSort` methods in the DDMIndexer class that creates the correct Sort configuration for the Nested and the Legacy cases.

The new `createDDMStructureFieldSort` methods are applied into the four places where we are sorting a search query with DDM fields:

1. AssetHelperImpl (used by the asset publisher and other asset-related functionality)
2. Data Engine (DataRecordResourceImpl.java)
3. Headless API (SortUtil.java)
4. Sort widget => new usage

For 1st, 2nd, and 3rd cases, I am replacing the existing direct search API usage with a call to the new createDDMStructureFieldSort method
In the fourth case, I am adding a new usage of this createDDMStructureFieldSort in order to solve the issue reported in this LPS.

Let me know if you have any question.

Thank you,
Jorge

### CODE REVIEW

**Comment by lipusz** [comment]
ci:abort

**Comment by lipusz** [comment]
ci:stop

**Comment by lipusz** [comment]
ci:stop

**Comment by lipusz** [comment]
ci:test:stable:stop

**Comment by lipusz** [comment]
ci:stop

**Comment by jorgediaz-lr** [comment]
According to latest Zsolt message, it is safe to use the CI, so I will execute the tests

@lipusz

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
ci:test:search

**Comment by lipusz** [comment]
:information_source: For Search related known CI results, please refer to [this](https://docs.google.com/spreadsheets/d/1cDzu4itg1XPtK4EyuwWB3MBwK3m7MpT3LqLcXltx9QA/edit?pli=1#gid=0) spreadsheet. This  can be accessed through https://search-ci.liferay.com/.

If you are not sure, feel free to reach out to the Search QA team in [#t-search-z-cont-integ](https://liferay.slack.com/archives/CLBEAPQ5D) or mention them through `@liferay/qa-search`.

Regards,
Tibi

**Comment by jorgediaz-lr** [comment]
About the **ci:test:search** execution, the failed test `LocalFile.Elasticsearch7EE#PortalClusterSmokeTest` is reported in the spreadsheet that Tibor has linked in his previous comment.

About the **ci:test:relevant** execution, the failed test `FragmentEntryProcessorFreemarkerTest.testProcessFragmentEntryLinkHTMLWithConfigurationItemSelectorJournalArticle` seems to be unrelated to this PR, nevertheless I will execute the ci:test:relevant tests again.

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
After executing the **ci:test:relevant** again:

- FragmentEntryProcessorFreemarkerTest.testProcessFragmentEntryLinkHTMLWithConfigurationItemSelectorJournalArticle error was not reproduced, so it seems unrelated to this pull
- There are two new failing tests LocalFile.AppBuilder#UnpublishAndRepublish and LocalFile.AppBuilder#DeployAppChoosingBetweenDifferentTableViews that weren't reproduced in previous ci:test:relevant run, so I think they are also unrelated. (note: second one is reported here: https://issues.liferay.com/browse/LRQA-59128)

**Comment by BryanEngler** [comment]
@jorgediaz-lr looks good to me, did you need to send it to any other teams for review?

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/portlet/shared/search/SortPortletSharedSearchContributor.java [hunk 1]
```diff
@@ -71,23 +72,33 @@ public void contribute(
...
+				throw new RuntimeException(portalException);
+			}
 		}
 
-		return sortBuilder.locale(
+		SortBuilder sortBuilder = _sortBuilderFactory.getSortBuilder();
+
+		return sortBuilder.field(
+			fieldValue
+		).sortOrder(
```
Comment (BryanEngler) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/portlet/shared/search/SortPortletSharedSearchContributor.java:99]: @jorgediaz-lr i think you should SF this in alphabetical order

**Comment by jorgediaz-lr** [comment]
Hi @BryanEngler 

Thanks for your review, I will fix the alphabetical order in the SortPortletSharedSearchContributor.java class and I will send it to the DDM team to double check that everything is ok, as I am moving all the code to the DDMIndexer class.

Regards,
Jorge

**Comment by jorgediaz-lr** [comment]
Sent to https://github.com/liferay-data-engine/liferay-portal/pull/255

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/portlet/shared/search/SortPortletSharedSearchContributor.java (hunk 1 above)
Comment (jorgediaz-lr) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/portlet/shared/search/SortPortletSharedSearchContributor.java:99]: I have added this commit in the new PR [`6e23b01` (#255)](https://github.com/liferay-data-engine/liferay-portal/pull/255/commits/6e23b01f7ca0477746745b428196f1f384979210)

**Pull request closed without merging**